FirebaseSharedSwift/Sources/third_party/FirebaseDataEncoder/FirebaseDataEncoder.swift

```diff
@@ -2118,7 +2118,8 @@ extension __JSONDecoder {
   fileprivate func unbox(_ value: Any, as type: Bool.Type) throws -> Bool? {
     guard !(value is NSNull) else { return nil }
 
-    if let number = value as? NSNumber {
+    let val = rcValBoolAdaptor(value)
+    if let number = val as? NSNumber {
```

Comment:
I believe all of the changes to this file would be unnecessary if it were possible to implement a cast from RemoteConfig to NSNumber that would apply on `as?`, but based on https://stackoverflow.com/questions/25585864/defining-explicit-conversion-for-custom-types-in-swift, it doesn't seem possible.